org.eclipse.lsp4e.jdt/src/org/eclipse/lsp4e/jdt/LSJavaProposal.java

```diff
@@ -94,18 +84,97 @@ public int getRelevance() {
 					break;
 				default:
 				}
-			}
 			relevanceComputed = true;
 		}
 		return relevance;
 	}
 	
-	private int computeBaseRelevance(LSCompletionProposal c) {
+	private int computeBaseRelevance() {
 		// Incorporate LSP4E category and rank into base relevance.
-		int base = MAX_BASE_RELEVANCE - (c.getRankCategory() - 1) * RANGE_WITHIN_CATEGORY;
-		int rank = c.getRankScore();
+		int base = MAX_BASE_RELEVANCE - (getRankCategory() - 1) * RANGE_WITHIN_CATEGORY;
+		int rank = getRankScore();
 		base -= (rank >= 0 && rank < RANGE_WITHIN_CATEGORY ? rank : RANGE_WITHIN_CATEGORY);
 		return base;
 	}
+	
+	/*
+	 * Copied from org.eclipse.jdt.internal.ui.text.java.AbstractJavaCompletionProposal.getInformationControlCreator()
+	 */
+	@Override
+	public @Nullable IInformationControlCreator getInformationControlCreator() {
+		Shell shell= JavaPlugin.getActiveWorkbenchShell();
+		if (shell == null || !BrowserInformationControl.isAvailable(shell))
+			return null;
+
+		if (infoControlCreator == null) {
+			/*
+			 * FIXME: Take control creators (and link handling) out of JavadocHover,
+			 * see https://bugs.eclipse.org/bugs/show_bug.cgi?id=232024
+			 */
+			IWorkbenchPart part = UI.getActivePart();
+			JavadocHover.PresenterControlCreator presenterControlCreator= new JavadocHover.PresenterControlCreator(part == null ? null : part.getSite());
+			infoControlCreator= new JavadocHover.HoverControlCreator(presenterControlCreator, true);
+		}
+		return infoControlCreator;
+	}
+
+	/*
+	 * Copied from org.eclipse.jdt.internal.ui.text.java.AbstractJavaCompletionProposal.getAdditionalProposalInfo(IProgressMonitor monitor)
+	 */
+	@Override
+	public Object getAdditionalProposalInfo(IProgressMonitor monitor) {
+		StringBuilder buffer = new StringBuilder((String) super.getAdditionalProposalInfo(monitor));
+		ColorRegistry registry= JFaceResources.getColorRegistry();
+		RGB fgRGB= registry.getRGB("org.eclipse.jdt.ui.Javadoc.foregroundColor"); //$NON-NLS-1$
+		RGB bgRGB= registry.getRGB("org.eclipse.jdt.ui.Javadoc.backgroundColor"); //$NON-NLS-1$
+		HTMLPrinter.insertPageProlog(buffer, 0, fgRGB, bgRGB, getCSSStyles());
+		HTMLPrinter.addPageEpilog(buffer);
+		return new JavadocBrowserInformationControlInput(null, null, buffer.toString(), 0);
+
+	}
+	
+	/**
+	 * Returns the style information for displaying HTML (Javadoc) content.
+	 * Copied from org.eclipse.jdt.internal.ui.text.java.AbstractJavaCompletionProposal.getCSSStyles()
+	 *
+	 * @return the CSS styles
+	 */
+	private @Nullable String getCSSStyles() {
+		if (fgCSSStyles == null) {
+			Bundle bundle= Platform.getBundle(JavaPlugin.getPluginId());
```

Comment:
Would better use `FrameworkUtil.getBundle(IJavaEditorTextHover.class)` (an API class related to the file we want)